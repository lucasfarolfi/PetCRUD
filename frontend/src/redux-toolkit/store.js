import {configureStore} from '@reduxjs/toolkit'

// import animalsReducer
export const store = configureStore({
    reducer: {
        animals: animalsReducer
    }
})