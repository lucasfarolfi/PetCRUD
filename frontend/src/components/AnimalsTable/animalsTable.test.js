import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import App from './index';
import TableLine, { convertDate } from './tableLine'
import {Router} from 'react-router-dom';
import store from '../../redux/store'
import {Provider} from 'react-redux'
import {createMemoryHistory} from 'history';

describe('Componente da tabela de animais', () =>{

    describe('Quando a tabela é renderizada', () =>{
        it('a linha da tabela deverá receber e exibir os atributos corretamente', () =>{
            const history = createMemoryHistory()

            const { container } = render(
                <Provider store={store}>
                    <Router history={history}><App>
                        <TableLine id="619572d1da346912a5ea67b7"
                            name="Jorge" date="05/10/2025" type="Cachorro" weight={15.5} />
                    </App></Router>
                </Provider>
            )
            expect(container.firstChild).toMatchSnapshot()
        })
    })
})
