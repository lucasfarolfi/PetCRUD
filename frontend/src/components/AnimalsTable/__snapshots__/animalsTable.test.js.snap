// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente da tabela de animais Quando a tabela é renderizada a linha da tabela deverá receber e exibir os atributos corretamente 1`] = `
<table
  cellspacing="0"
  class="table"
  data-testid="animals-table"
>
  <thead>
    <th>
      Data de nascimento
    </th>
    <th>
      Nome
    </th>
    <th>
      Tipo
    </th>
    <th>
      Peso
    </th>
    <th>
      Alterar
    </th>
    <th>
      Excluir
    </th>
  </thead>
  <tbody>
    <tr>
      Nenhum projeto encontrado
    </tr>
  </tbody>
</table>
`;
